### DISCUSSION TITLE
Some suggestions for the paint editor -- Layers, corners and more

### DISCUSSION THREAD

**Discussion by LercDsgn** [ScratchAddons/ScratchAddons#3088] [Ideas]
### All Suggestions: 
_Layer Borders, Layer Visualizing, Corner Point Binding, Move Around Canvas With Hold & Drag_

---

**1. Layer borders**
Ever wanted to get a shape under the border of another shape but not the shape itself? Now you can do this!

**2. Layer Visualizing**
Easily view, switch between and change the order of layers.

**3. Corner Point Binding (Available while the Corner Adjusting Tool is selected)**
Binding corners  are similar to grouping elements on the editor. When you take an action on one, all the connected corners are going to reflect.

**4. Move Around Canvas With Hold & Drag**
With this one, you're no longer stuck at one part of the canvas, zoomed in and trying to move the view a little. Just press, hold, move and ta-da!

Edit, BONUS(-es): 
- **Instantly Draw A Line Between Two Points (Available while the Corner Adjusting Tool is selected)**, 
- **Change Properties (color, layer etc.) Of Area That Selected Points Of A Shape Formed (seperately from the rest of the shape)**.

**Comment by fierin** [comment]
This would go handin hand with onion skinning to make an amazing and intuative experience for people used to drawing apps with similar features.

Similarly i want to see some way for animation interpolation (more advances sujjestion)

**Reply by LercDsgn** [reply]
There could be a spread feature to spread the shape with (slightly) different sizes, rotations, colors etc., to fill up a selected area. The spread rate (how random the spread will be) and spread options (what and how the spread tool should spread about the shape) could be helpful as well. And it may also create a new object to easily play with the new spread shapes. This can also be used on areas, which will spread the selected area. A little off but all the spread options, including spread rate (?), might have an option to randomize the value of itself.

Edit: After all these, we really need some blocks to go along with paint editor features.
